driver-core/src/test/java/com/datastax/driver/core/HostConnectionPoolTest.java

```diff
@@ -205,9 +231,9 @@ public void should_adjust_connection_keyspace_on_dequeue_if_pool_state_is_differ
             // will fail, and thus the enqueued connection futures will fail in the following ways:
             // 1. DriverInternalError for set keyspace returning 'ROWS' response
             // 2. TransportException for connection being closed, when this happens we stop checking these futures.
-            // 3. Never completing because no connections are available.
+            // 3. Timing out because no connections are available.
```

Comment:
D'oh, it should have been apparent to me that this could be a problem.  Even documented that the future would never complete here!